### PR TITLE
feat: Include whitelisted HTTP headers in extended_unique_key computation.

### PR DESCRIPTION
**Pull request by AkhilProto**
### Description

This pull request enhances the `compute_unique_key` function in the `src/crawlee/_utils/requests.py` file to include HTTP headers in the unique key computation and adds corresponding unit tests. The most important changes include adding new parameters for headers and whitelisted headers, modifying the logic to compute the unique key, and adding tests to verify the new functionality.

Enhancements to `compute_unique_key` function:

* [`src/crawlee/_utils/requests.py`](diffhunk://#diff-3c15fd2da12b7729eb02594536c318224566fac913743ec91296f989b1d9c2edR94-R110): Added `headers` and `whitelisted_headers` parameters to the `compute_unique_key` function to include HTTP headers in the unique key computation.
* [`src/crawlee/_utils/requests.py`](diffhunk://#diff-3c15fd2da12b7729eb02594536c318224566fac913743ec91296f989b1d9c2edL120-L140): Modified the logic to compute the unique key by including whitelisted headers if provided.

### Issues

- Closes: #548 

### Testing

* [`tests/unit/_utils/test_compute_unique_key.py`](diffhunk://#diff-3f240e787e5702e07f327a6a70f9cb9961e1104fba6468aa1e2baa7befec7643R1-R18): Added unit tests to verify the new functionality of including headers in the unique key computation using `pytest`.

### Checklist

- [X] CI passed

### CODE REVIEW

**Review by vdusek — CHANGES_REQUESTED** [review]
Thanks for the contribution. Could you please update your code so that CI passes? Check [CONTRIBUTING.md](https://github.com/apify/crawlee-python/blob/master/CONTRIBUTING.md) for more information.

**Comment by AkhilProto** [comment]
Sure. I'll start working on it.

**Comment by AkhilProto** [comment]
Hey! Even after making the necessary changes and making sure that there is no room for errors, the CI tests keep failing.Upon closer inspection of the logs, it seems that the errors are originating from other files which I haven't touched. Any help regarding this would be appreciated.
Thanks!

**Comment by AkhilProto** [comment]
Hey @vdusek.. Waiting for a review.

**Comment by vdusek** [comment]
Hey @AkhilProto, CI is still not passing.

**Comment by AkhilProto** [comment]
Yup! Had to put some effort in reformatting the other files too.. All checks have passed!

File: website/generate_module_shortcuts.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def get_module_shortcuts(module, parent_classes=None):
     return shortcuts
```
Comment (vdusek) [inline website/generate_module_shortcuts.py:25]: Please do not touch this file.

**Review by vdusek — CHANGES_REQUESTED** [review]
Could you also write some tests please?

**Comment by AkhilProto** [comment]
Added a test and also reverted the changes made in 'website/generate_module_shortcuts.py'

File: src/crawlee/_utils/requests.py [hunk 2]
```diff
@@ -91,19 +91,23 @@ def compute_unique_key(
     *,
     keep_url_fragment: bool = False,
     use_extended_unique_key: bool = False,
+    headers: dict[str, str] | None = None,
```
Comment (vdusek) [inline src/crawlee/_utils/requests.py:94]: Could you please use [HttpHeaders](https://github.com/apify/crawlee-python/blob/v0.3.8/src/crawlee/_types.py#L43) type for headers? It should also help with the normalization.

**Review by vdusek — CHANGES_REQUESTED** [review]

**Comment by vdusek** [comment]
> Added a test and also reverted the changes made in 'website/generate_module_shortcuts.py'

I don't see it :slightly_smiling_face:.

**Comment by AkhilProto** [comment]
> > Added a test and also reverted the changes made in 'website/generate_module_shortcuts.py'
> 
> I don't see it 🙂.

I did update `test_requests.py`

**Comment by AkhilProto** [comment]
mb.. i realized i had the tests saved locally :sweat_smile: ... i have done it now

**Comment by vdusek** [comment]
Sorry, but I have to close it. There was another contributor who managed to solve it faster - #609.

**Comment by AkhilProto** [comment]
Fair enough.. It's been a pretty good experience tho! Thanks!

**Pull request closed without merging**